lib/dispatcher.rb

```diff
@@ -35,6 +35,15 @@ def deliver_new_proposal_emails(proposal)
     self.email_sent_confirmation(proposal)
   end
 
+  def deliver_attachment_emails(proposal)
+    proposal.users.each do |user|
+      # do not send email to approvers who have not yet heard about the proposal
+      approval = proposal.approvals.find_by(user_id: user.id)
+      next if approval && approval.pending?
```

Comment:
Can we merge this with the two or three other ways to determine which users should be notified? I'm thinking of the methods on comments and on updates to a proposal.